charts/pulsar/templates/grafana-pvc.yaml

```diff
@@ -0,0 +1,40 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+{{- if or .Values.monitoring.grafana .Values.extra.monitoring }}
+{{- if and (and .Values.persistence .Values.volumes.persistence) .Values.grafana.volumes.persistence }}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: "{{ template "pulsar.fullname" . }}-{{ .Values.grafana.component }}-{{ .Values.grafana.volumes.data.name }}"
+  namespace: {{ template "pulsar.namespace" . }}
+spec:
+  resources:
+    requests:
+      storage: {{ .Values.grafana.volumes.data.size }}
+  accessModes: [ "ReadWriteOnce" ]
+{{- if .Values.grafana.volumes.data.storageClassName }}
+  storageClassName: "{{ .Values.grafana.volumes.data.storageClassName }}"
+{{- else if and (not (and .Values.volumes.local_storage .Values.grafana.volumes.data.local_storage)) .Values.grafana.volumes.data.storageClass }}
+  storageClassName: "{{ template "pulsar.fullname" . }}-{{ .Values.grafana.component }}-{{ .Values.grafana.volumes.data.name }}"
```

Comment:
other components create their own storageclass so we can use this storageclass, seem that's not true with grafana here, then we should either remove this or add creation for storageclass.

Reply:
Add creation for grafana storageclass.